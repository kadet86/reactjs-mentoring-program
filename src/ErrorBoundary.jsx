// @flow
import * as React from 'react';

type Props = {
    children: React.Node,
};
type State = {
    error?: Error,
};

export default class ErrorBoundary extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {};
    }

    static getDerivedStateFromError(error: Error) {
        return { error };
    }

    componentDidCatch(error: Error, info: any) {
        console.error(error, info);
    }

    render() {
        if (this.state.error) {
            return <h1>Something went wrong...</h1>;
        }

        return this.props.children;
    }
}
