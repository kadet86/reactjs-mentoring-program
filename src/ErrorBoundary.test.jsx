import React from 'react';
import {shallow, mount} from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary component', () => {
    it('has persistent regular snapshot', () => {
        const component = shallow((
            <ErrorBoundary>
                <div id="1"></div>
                <div id="2"></div>
            </ErrorBoundary>
            )
        );
        expect(component).toMatchSnapshot();
    });

    it('has persistent error snapshot', () => {
        const TestError = () => {throw 'error';};
        console.error = jest.fn();
        const component = mount((
            <ErrorBoundary>
                <TestError />
            </ErrorBoundary>
            )
        );

        expect(console.error).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });
});