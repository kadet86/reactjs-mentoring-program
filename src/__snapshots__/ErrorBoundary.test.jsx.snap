// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component has persistent error snapshot 1`] = `ReactWrapper {}`;

exports[`ErrorBoundary component has persistent regular snapshot 1`] = `ShallowWrapper {}`;
